### ISSUE TITLE
Creating <svg> inside of an existing element (i.e., <div>)

### ISSUE THREAD

**Issue by spaddlewit** [anseki/leader-line#300]
Instead of the library appending <svg> paths to <body>, is there a way to have it append to a <div>? I want to use it in combination with PanZoom ( https://github.com/anvaka/panzoom ), which works at the element-level. I tried calling line.position() on a pan/zoom, but it looks like leader-line isn't aware of element transforms.

**Comment by anseki** [comment]
Hi @spaddlewit, thank you for the comment.
You can move the SVG element to another element from `<body>` if you really want it.
For example:
```js
document.getElementById('your-container').appendChild(document.querySelector('.leader-line'));
```
https://github.com/anseki/leader-line/issues?q=is%3Aissue+querySelector
Note that it is not supported, and it is deprecated.

**Comment by spaddlewit** [comment]
Thank you! Although with some testing, it appears I just need to call the position() method of each line whenever the panzoom library does a transform. But when I do it, it says 'line.position' is not a valid function.

instance.on('transform', function(e) {
  // This event will be called along with events above.
  console.log('Fired when any transformation has happened', e);

var lines = document.querySelectorAll('.leader-line');
[].forEach.call(lines, function(line) {
line.position();
});
});

**Comment by anseki** [comment]
Yes, of course it doesn't have `position` method because that is not LeaderLine.
The `querySelectorAll` method returns `NodeList`. Also, that is not a method of LeaderLine, that is DOM API.
See document: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll

**Comment by spaddlewit** [comment]
Sorry, I am not the best with a loosely-typed language like Javascript.
Do I need to maintain my own list of the leader lines, adding each one as I create one with a constructor and then iterate that, or is there some way to query all of the leader lines in the document?

**Comment by anseki** [comment]
Sorry, my English is poor.
Do you mean that you want to know how to use `Array` in JavaScript?

**Comment by spaddlewit** [comment]
Thank you, but no - I was wondering if there was a way to query the document for all '.leader-line' items to call position() on them. I did get the Array technique to work, though.

**Comment by anseki** [comment]
That CSS selector gets all SVG elements of LeaderLine. Note that those are not LeaderLine.
See: https://github.com/anseki/leader-line/issues/300#issuecomment-1050496116
Anyway, calling `querySelector/querySelectorAll` method every time you want to do something is a bad choice because those method are costly, and it may make your app slow.
See: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector
Also, generally speaking, that seems to be strange logic in basic computer programing.

**Comment by spaddlewit** [comment]
Yes, I agree. I figured I could optimize later.
Thank you for the extra information, even though we are starting to get off topic! :)
